### PR TITLE
[SD-3670] fixed issues relate to interactive macros

### PR DESCRIPTION
**Pull request by mdhaman**
- diff returned from the server was not clear after the item is closed and was visible if a new item is opened.
- replace was not highlighting text.

### CODE REVIEW

**Comment by mdhaman** [comment]
Depends on [PR-211](https://github.com/superdesk/superdesk-core/pull/211)

File: scripts/superdesk-authoring/macros/macros.js [hunk 1]
```diff
@@ -82,11 +81,13 @@ function MacrosReplaceDirective(macros, editor) {
...
+            scope.diff = null;
 
-            scope.$watch(function() {
-                return macros.diff;
-            }, function(diff) {
+            scope.$on('macro:diff', function(evt, diff) {
                 scope.diff = diff;
+            });
+
+            scope.$watch('diff', function(diff) {
```
Comment (petrjasek) [inline scripts/superdesk-authoring/macros/macros.js]: seems like this watch can be avoided altogether

**Comment by petrjasek** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

**Pull request merged**